### DISCUSSION TITLE
How would you implement a gift card table with RLS?

### DISCUSSION THREAD

**Discussion by cloudorbush** [supabase/supabase#6912] [Questions] [answered]
Say I have a `gift_cards` table.

The `gift_cards` table contains two columns:

`secret_code` and `balance`.

How would you implement a RLS policy which enables anon users to read the `balance` of a row on the `gift_cards` table only if they know the matching `secret_code`?

Any insight is greatly appreciated.

**Comment by GaryAustin1** [comment, marked as answer]
There may be other ways, but I would lock the table with RLS to not allow selects.  Then you can create an rpc function with "security definer" type to bypass RLS.  Pass that function your secret code and do the table lookup in the function and return your balance or row.
